There is a stack trace in the test logs when the test daemons are trying to start. Because of this stacktrace, the tests don't run and the build times out.

```
04:12:09 Traceback (most recent call last):
04:12:09   File "/testing/tests/runtests.py", line 411, in <module>
04:12:09     main()
04:12:09   File "/testing/tests/runtests.py", line 396, in main
04:12:09     status = parser.run_integration_tests()
04:12:09   File "/testing/tests/runtests.py", line 316, in run_integration_tests
04:12:09     with TestDaemon(self):
04:12:09   File "/testing/tests/integration/__init__.py", line 173, in __enter__
04:12:09     self.start_zeromq_daemons()
04:12:09   File "/testing/tests/integration/__init__.py", line 230, in start_zeromq_daemons
04:12:09     minion = salt.minion.Minion(self.minion_opts)
04:12:09   File "/testing/salt/minion.py", line 606, in __init__
04:12:09     opts['environment'],
04:12:09   File "/testing/salt/pillar/__init__.py", line 91, in compile_pillar
04:12:09     ret = self.sreq.crypted_transfer_decode(load, tries=3, timeout=7200)
04:12:09   File "/testing/salt/transport/__init__.py", line 252, in crypted_transfer_decode
04:12:09     return pcrypt.loads(ret)
04:12:09   File "/testing/salt/crypt.py", line 797, in loads
04:12:09     data = self.decrypt(data)
04:12:09   File "/testing/salt/crypt.py", line 769, in decrypt
04:12:09     sig = data[-self.SIG_SIZE:]
04:12:09 TypeError: unhashable type
```

--- a/salt/transport/__init__.py
+++ b/salt/transport/__init__.py
@@ -242,9 +242,13 @@ class ZeroMQChannel(Channel):
     def crypted_transfer_decode_dictentry(self, load, dictkey=None, tries=3, timeout=60):
         ret = self.sreq.send('aes', self.auth.crypticle.dumps(load), tries, timeout)
         key = self.auth.get_keys()
-        aes = key.private_decrypt(ret['key'], 4)
-        pcrypt = salt.crypt.Crypticle(self.opts, aes)
-        return pcrypt.loads(ret[dictkey])
+        try:
+            aes = key.private_decrypt(ret['key'], 4)
+        except (TypeError, KeyError):
+            return None
+        else:
+            pcrypt = salt.crypt.Crypticle(self.opts, aes)
+            return pcrypt.loads(ret[dictkey])
 
     def _crypted_transfer(self, load, tries=3, timeout=60):
         '''

